hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHADBTransactionBufferImpl.java

```diff
@@ -138,5 +138,8 @@ public String toString() {
 
   @Override
   public void close() throws IOException {
+    if (currentBatchOperation != null) {
+      currentBatchOperation.close();
```

Comment:
Can `currentBatchOperation` be safely closed twice without any side effects? If not, does it make sense to set `currentBatchOperation` to `null` here just in case there is a pathological caller where this could be closed twice?

```suggestion
      currentBatchOperation.close();
      currentBatchOperation = null;
```

Reply:
`currentBatchOperation` can be closed multiple times without side effects and only the first invoke makes an impact. Internally, every RocksObject maintains an atomic internal state to tell if it's closed.

If we put a side-effect to change the `SCMHADBTransactionBufferImpl` state in `close`, we may need to make it `synchronized`.

Reply:
Got it. Thx!